### PR TITLE
support statistics on replicated merge tree

### PR DESCRIPTION
**Pull request by hanfei1991**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support statistics with ReplicatedMergeTree

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 823a7d37f3bab8d4aa9cb8b2ff8b7b087e8a3037 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64934/823a7d37f3bab8d4aa9cb8b2ff8b7b087e8a3037/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_manipulate_statistics/test.py [hunk 1]
```diff
@@ -122,3 +126,32 @@ def test_single_node_normal(started_cluster):
     """
     )
     run_test_single_node(started_cluster)
+
+
+def test_replicated_table_ddl(started_cluster):
```
Comment (antaljanosbenjamin) [inline tests/integration/test_manipulate_statistics/test.py:131]: Maybe create a tests where you modify the columns (e.g.: rename, change type to a valid one, change type to invalid type, drop column, etc) and verify that CH can handle it without issues.

File: src/Storages/ColumnsDescription.cpp [hunk 2]
```diff
@@ -207,6 +207,9 @@ void ColumnDescription::readText(ReadBuffer & buf)
 
             if (col_ast->settings)
                 settings = col_ast->settings->as<ASTSetQuery &>().changes;
+
+            if (col_ast->statistics_desc)
```
Comment (antaljanosbenjamin) [inline src/Storages/ColumnsDescription.cpp:211]: I don't really get the followings, probably because of my limited knowledge:
- why this change makes statistics work with RMT?
- why the call to `ColumnStatisticsDescription::fromColumnDeclaration` in `InterpreterCreateQuery` isn't enough to make statistics work with RMT?

File: tests/integration/test_manipulate_statistics/test.py (hunk 1 above)
Comment (hanfei1991) [reply tests/integration/test_manipulate_statistics/test.py:131]: Goed idee! on it (actually I found new bugs :(

Comment (hanfei1991) [reply tests/integration/test_manipulate_statistics/test.py:131]: done, PTAL @antaljanosbenjamin

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by hanfei1991** [comment]
flaky test https://github.com/ClickHouse/ClickHouse/issues/64790
